components/ShadowLayer/src/MDCShadowLayer.m

```diff
@@ -304,25 +334,87 @@ - (void)commonLayoutSublayers {
   _topShadow.bounds = bounds;
 
   if (_shadowMaskEnabled) {
-    _bottomShadow.mask = [self shadowLayerMaskForLayer:_bottomShadow];
-    _topShadow.mask = [self shadowLayerMaskForLayer:_topShadow];
+    [self configureShadowLayerMaskForLayer:_topShadowMask];
+    [self configureShadowLayerMaskForLayer:_bottomShadowMask];
   }
   // Enforce shadowPaths because otherwise no shadows can be drawn. If a shadowPath
   // is already set, use that, otherwise fallback to just a regular rect because path.
-  if (!_bottomShadow.shadowPath) {
+  if (!_bottomShadow.shadowPath || _shadowPathIsInvalid) {
     if (self.shadowPath) {
       _bottomShadow.shadowPath = self.shadowPath;
     } else {
       _bottomShadow.shadowPath = [self defaultShadowPath].CGPath;
     }
   }
-  if (!_topShadow.shadowPath) {
+  if (!_topShadow.shadowPath || _shadowPathIsInvalid) {
     if (self.shadowPath) {
       _topShadow.shadowPath = self.shadowPath;
     } else {
       _topShadow.shadowPath = [self defaultShadowPath].CGPath;
     }
   }
+  _shadowPathIsInvalid = NO;
+}
+
+@end
+
+@implementation MDCAnimatedShapeLayer
+
+- (id<CAAction>)actionForKey:(NSString *)key {
+  if ([key isEqualToString:@"path"] || [key isEqualToString:@"shadowPath"]) {
+    CABasicAnimation *animation = [CABasicAnimation animationWithKeyPath:key];
```

Comment:
Did you test this using the spring animation APIs? I'm curious whether we should be copying the bounds animation (so that we also spring these properties) or copying its properties (as you're doing now).

Reply:
The existing behavior won't work with springs. I'm testing copying spring animations and it isn't working as expected so I'll have to explore that a bit more. Copying may be better - I don't have a strong preference either way.

Reply:
May be ok to punt on that for now. This PR is an improvement over the existing behavior as-is.